vsb/databases/opensearch/opensearch.py

```diff
@@ -0,0 +1,207 @@
+            )
+            self.client.indices.create(index=self.index_name, body=index_body)
+            self.created_index = True
+            time.sleep(30)
+        else:
+            logger.info(
+                f"OpenSearchDB: Index '{self.index_name}' already exists. Skipping index creation."
+            )
+            self.created_index = False
+
+    def get_batch_size(self, sample_record: Record) -> int:
+        # Similar constraints as Pinecone, OpenSearch also has limits on batch sizes
+        # max_record_size = 1024 * 40  # Estimate 40KB for each record
+        # max_batch_size = 500  # OpenSearch handles bulk requests in chunks
+        # For now, we'll use a batch size of 100
+        batch_size = 100
+        return batch_size
+
+    def get_namespace(self, namespace: str) -> Namespace:
+        return OpenSearchNamespace(
+            self.client, self.index_name, self.dimensions, namespace
+        )
+
+    def initialize_population(self):
+        if self.skip_populate:
+            return
+        if not self.created_index and not self.overwrite:
+            msg = (
+                f"OpenSearchDB: Index '{self.index_name}' already exists - cowardly "
+                f"refusing to overwrite existing data. Specify --overwrite to "
+                f"delete it, or specify --skip_populate to skip population phase."
+            )
+            logger.critical(msg)
+            raise StopUser()
+        if not self.created_index:
+            try:
+                logger.info(
+                    f"OpenSearchDB: Deleting existing index '{self.index_name}' before "
+                    f"population (--overwrite=True)"
+                )
+                self.client.indices.delete(index=self.index_name)
+                logger.info(f"Index '{self.index_name}' cleared for population")
+                time.sleep(10)
+                self.create_index()
+            except Exception as e:
+                logger.critical(f"Error deleting index '{self.index_name}': {e}")
+                raise StopUser()
+
+    def finalize_population(self, record_count: int):
+        """Wait until all records are visible in the index"""
+        logger.debug(f"OpenSearchDB: Waiting for record count to reach {record_count}")
+        with vsb.logging.progress_task(
+            "  Finalize population", "  ✔ Finalize population", total=record_count
+        ) as finalize_id:
+            while True:
+                index_count = self.client.count(index=self.index_name)["count"]
+                if vsb.progress:
+                    vsb.progress.update(finalize_id, completed=index_count)
+                if index_count >= record_count:
+                    logger.debug(
+                        f"OpenSearchDB: Index vector count reached {index_count}, "
+                        f"finalize is complete"
+                    )
+                    break
+                time.sleep(5)
```

Comment:
Probably fine to reduce this sleep to say 1 second; otherwise at the moment there's at-most 5s additional time needed to be waited if the client.count() call _just_ missed having all records present.